tests/e2e-instrumentation/instrumentation-dotnet-musl/01-install-app.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment-with-sidecar
+spec:
+  selector:
+    matchLabels:
+      app: my-pod-with-sidecar
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: my-pod-with-sidecar
+      annotations:
+        sidecar.opentelemetry.io/inject: "true"
+        instrumentation.opentelemetry.io/inject-dotnet: "true"
+        instrumentation.opentelemetry.io/otel-dotnet-auto-runtime: "linux-musl-x64"
+    spec:
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 3000
+        fsGroup: 2000
+      containers:
+      - name: myapp
+        image: docker.io/avadhutp123/aspnetapp@sha256:d2e8d3415f6f12efae0369aa0a9777a58583841fb133f33e10a73adb9fb392da  # source code of the application: https://github.com/dotnet/dotnet-docker/tree/main/samples/aspnetapp
```

Comment:
Is there an official image available? 

We are discussing about using official apps that are well maintained or hosting the test apps in this repo.

cc) @iblancasa

Reply:
Also if we decide to use an image it should be published for multiple arch: arm, amd64, IBM Z/P

Reply:
```
avadhutp123/aspnetapp          <none>              8606880cf2e0   11 months ago   220MB
```

Reply:
In e2e tests in fact you are not testing applications. It can be even empty image. E2E tests are checking only if appropriate env. variables are injected.

For now, .NET auto supports linux-x64 (glibc/amd64) and linux-musl-x64 (musl/amd64). ARM (it can be supported), IBM Z/P (I am not sure if it even can be supported) are not supported by auto-instrumentation

Reply:
> In e2e tests in fact you are not testing applications. It can be even empty image. E2E tests are checking only if appropriate env. variables are injected.

This is totally true. I think we may need to revisit the current approach to use an echo server or something similar.

> For now, .NET auto supports linux-x64 (glibc/amd64) and linux-musl-x64 (musl/amd64). ARM (it can be supported), IBM Z/P (I am not sure if it even can be supported) are not supported by auto-instrumentation

Yes, they cannot be supported.

> EDIT: I am not introducing here anything new, it was used also by glibc test.

Yes, this change is fine.

Reply:
> > In e2e tests in fact you are not testing applications. It can be even empty image. E2E tests are checking only if appropriate env. variables are injected.
> 
> This is totally true. I think we may need to revisit the current approach to use an echo server or something similar.
> 
To have full e2e coverage, tt will be great to check if some auto-instrumented (for .NET - using bytecode) traces/metrics/logs are generated by the application.
.NET auto-instrumentation should silently continue in case of non-working correctly. The same approach is recommended for other OTel-auto versions.

For sure another PR.

Reply:
Ok so, I have some PRs to minimize the applications, add support to multiple architectures and hos them in this repository. Adding a full E2E test can be a follow up of that.